### PR TITLE
Jormungandr Kraken: add new param into database and api

### PR DESCRIPTION
**Pull request by xlqian**
http://jira.canaltp.fr/browse/NAVP-419

add max_duration
add walking_transfer_penalty
add params for filtering night bus

into api and db

### CODE REVIEW

File: source/jormungandr/jormungandr/interfaces/v1/Journeys.py [hunk 1]
```diff
@@ -587,6 +588,10 @@ def __init__(self):
         parser_get.add_argument("traveler_type", type=option_value(acceptable_traveler_types))
         parser_get.add_argument("_override_scenario", type=str, description="debug param to specify a custom scenario")
 
+        parser_get.add_argument("walking_transfer_penalty", type=int)
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/interfaces/v1/Journeys.py]: Do we really want to expose this params?

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 2]
```diff
@@ -139,6 +139,8 @@ def create_pb_request(requested_type, request, dep_mode, arr_mode):
     if "details" in request and request["details"]:
         req.journeys.details = request["details"]
 
+    req.journeys.walking_transfer_penalty = request['walking_transfer_penalty']
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/scenarios/new_default.py]: This need to be done in the default scenario too

File: source/tyr/tyr/resources.py [hunk 3]
```diff
@@ -335,30 +339,40 @@ def put(self, id=None, name=None):
                 help='', location=('json', 'values'),
                 default=instance.max_duration_fallback_mode)
 
+        parser.add_argument('max_duration', type=int, help='latest time point of research', location=('json', 'values'),
```
Comment (kinnou02) [inline source/tyr/tyr/resources.py]: and not the others?

File: source/jormungandr/jormungandr/interfaces/v1/Journeys.py (hunk 1 above)
Comment (antoine-de) [inline source/jormungandr/jormungandr/interfaces/v1/Journeys.py]: I don't think so, for me they are only instances params (or maybe hidden ones for test purpose ?)

File: source/tyr/tyr/resources.py (hunk 3 above)
Comment (xlqian) [inline source/tyr/tyr/resources.py]: oops

File: source/jormungandr/jormungandr/scenarios/new_default.py (hunk 2 above)
Comment (xlqian) [inline source/jormungandr/jormungandr/scenarios/new_default.py]: ok

File: source/jormungandr/jormungandr/interfaces/v1/Journeys.py (hunk 1 above)
Comment (xlqian) [inline source/jormungandr/jormungandr/interfaces/v1/Journeys.py]: according to the doc http://confluence.canaltp.fr/pages/viewpage.action?pageId=8620368, these params are exposed, we can expose them just for debug purpose?

Comment (kinnou02) [inline source/jormungandr/jormungandr/interfaces/v1/Journeys.py]: If we expose them we need to prefix them with an underscore

Comment (xlqian) [inline source/jormungandr/jormungandr/interfaces/v1/Journeys.py]: Ok!

File: source/navitiacommon/navitiacommon/default_values.py [hunk 4]
```diff
@@ -80,6 +80,17 @@
 #only journey with this mode (or slower mode) will be used for calculate the max_duration
 max_duration_fallback_mode = 'walking'
 
+# specify the latest time point of request, relative to datetime, in second
+max_duration = 60*60*24  # seconds
+
+# the penalty for each walking transfer
+walking_transfer_penalty = 2  # minutes
```
Comment (kinnou02) [inline source/navitiacommon/navitiacommon/default_values.py]: All other duration are in seconds, there is a specific reason to use minutes for this one?

File: source/tyr/tyr/resources.py [hunk 5]
```diff
@@ -335,30 +339,49 @@ def put(self, id=None, name=None):
...
+        parser.add_argument('max_duration', type=int, help='latest time point of research, in second',
+                            location=('json', 'values'), default=instance.max_duration)
+
+        parser.add_argument('walking_transfer_penalty', type=int, help='transfer penalty, in min',
+                            location=('json', 'values'), default=instance.walking_transfer_penalty)
+
+        parser.add_argument('night_bus_filter_max_factor', type=int, help='night bust filter param',
+                            location=('json', 'values'), default=instance.night_bus_filter_max_factor)
+
+        parser.add_argument('night_bus_filter_base_factor', type=int, help='night bust filter param',
```
Comment (kinnou02) [inline source/tyr/tyr/resources.py]: typo: bus not bust, same on the previous one

File: source/navitiacommon/navitiacommon/default_values.py (hunk 4 above)
Comment (xlqian) [inline source/navitiacommon/navitiacommon/default_values.py]: you're right...

**Comment by xlqian** [comment]
Jenkins, test this please

File: source/tyr/tyr/resources.py [hunk 6]
```diff
@@ -344,30 +348,49 @@ def put(self, id=None, name=None):
                 help='', location=('json', 'values'),
                 default=instance.max_duration_fallback_mode)
 
+        parser.add_argument('max_duration', type=int, help='latest time point of research, in second',
+                            location=('json', 'values'), default=instance.max_duration)
+
+        parser.add_argument('walking_transfer_penalty', type=int, help='transfer penalty, in min',
```
Comment (antoine-de) [inline source/tyr/tyr/resources.py]: in seconds

File: source/tyr/tyr/resources.py [hunk 7]
```diff
@@ -344,30 +348,49 @@ def put(self, id=None, name=None):
...
-            instance.min_tc_with_bike = args['min_tc_with_bike']
-            instance.min_tc_with_bss = args['min_tc_with_bss']
-            instance.min_bike = args['min_bike']
-            instance.min_bss = args['min_bss']
-            instance.min_car = args['min_car']
-            instance.min_duration_too_long_journey = args['min_duration_too_long_journey']
-            instance.factor_too_long_journey = args['factor_too_long_journey']
-            instance.max_duration_criteria = args['max_duration_criteria']
-            instance.max_duration_fallback_mode = args['max_duration_fallback_mode']
+            def map_args_to_instance(attr_name):
```
Comment (antoine-de) [inline source/tyr/tyr/resources.py:366]: :+1:

**Comment by xlqian** [comment]
Jenkins, test this please

**Comment by xlqian** [comment]
Jenkins, test this please

File: source/fare/tests/fare_integration_test.cpp [hunk 8]
```diff
@@ -161,7 +162,7 @@ BOOST_AUTO_TEST_CASE(test_protobuff_no_data) {
 
     georef::StreetNetwork sn_worker(*b.data->geo_ref);
     pbnavitia::Response resp = make_response(raptor, origin, destination, {test::to_posix_timestamp("20120614T080000")},
-                                             true, type::AccessibiliteParams(), {}, sn_worker, type::RTLevel::Base);
+                                             true, type::AccessibiliteParams(), {}, sn_worker, type::RTLevel::Base, 2_min);
```
Comment (pbougue) [inline source/fare/tests/fare_integration_test.cpp]: line length

Comment (antoine-de) [inline source/fare/tests/fare_integration_test.cpp]: the relou

**Comment by xlqian** [comment]
Jenkins, test this please!

File: source/jormungandr/jormungandr/interfaces/v1/Journeys.py [hunk 9]
```diff
@@ -573,7 +574,7 @@ def __init__(self):
         parser_get.add_argument("data_freshness",
                                 type=option_value(['base_schedule', 'adapted_schedule', 'realtime']))
 # a supprimer
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/Journeys.py]: what is there to delete?

File: source/fare/tests/fare_integration_test.cpp (hunk 8 above)
Comment (pbougue) [inline source/fare/tests/fare_integration_test.cpp]: Il l'avait fait au-dessus!
Nazi j'veux bien, mais relou... :)

**Comment by antoine-de** [comment]
Jenkins, test this please

**Pull request merged**